marlowe-playground-client/src/MarloweEditor/State.purs

```diff
@@ -0,0 +1,222 @@
+import StaticData as StaticData
+import Text.Pretty (pretty)
+import Types (WebData)
+
+handleAction ::
+  forall m.
+  MonadAff m =>
+  SPSettings_ SPParams_ ->
+  Action ->
+  HalogenM State Action ChildSlots Void m Unit
+handleAction _ (ChangeKeyBindings bindings) = do
+  assign _keybindings bindings
+  void $ query _marloweEditorPageSlot unit (Monaco.SetKeyBindings bindings unit)
+
+handleAction _ (HandleEditorMessage (Monaco.TextChanged text)) = do
+  modify_
+    ( set _selectedHole Nothing
+        <<< set _hasUnsavedChanges' true
+    )
+  liftEffect $ LocalStorage.setItem marloweBufferLocalStorageKey text
+  analysisState <- use _analysisState
+  let
+    parsedContract = parseContract text
+
+    unreachableContracts = getUnreachableContracts analysisState
+
+    (Tuple markerData additionalContext) = Linter.markers unreachableContracts parsedContract
+  markers <- query _marloweEditorPageSlot unit (Monaco.SetModelMarkers markerData identity)
+  void $ traverse editorSetMarkers markers
+  {-
+    There are three different Monaco objects that require the linting information:
+      * Markers
+      * Code completion (type aheads)
+      * Code suggestions (Quick fixes)
+     To avoid having to recalculate the linting multiple times, we add aditional context to the providers
+     whenever the code changes.
+  -}
+  providers <- query _marloweEditorPageSlot unit (Monaco.GetObjects identity)
+  case providers of
+    Just { codeActionProvider: Just caProvider, completionItemProvider: Just ciProvider } -> pure $ updateAdditionalContext caProvider ciProvider additionalContext
+    _ -> pure unit
+
+handleAction _ (HandleDragEvent event) = liftEffect $ FileEvents.preventDefault event
+
+handleAction settings (HandleDropEvent event) = do
+  liftEffect $ FileEvents.preventDefault event
+  contents <- liftAff $ readFileFromDragEvent event
+  void $ editorSetValue contents
+  -- updateContractInState contents --FIXME
+  -- FIXME: check but I don't think this is needed after the split
+  -- setOraclePrice settings
+  -- pure unit is only here so the comment stays in place
+  pure unit
+
+handleAction _ (MoveToPosition lineNumber column) = do
+  void $ query _marloweEditorPageSlot unit (Monaco.SetPosition { column, lineNumber } unit)
+
+handleAction settings (LoadScript key) = do
+  for_ (preview (ix key) StaticData.marloweContracts) \contents -> do
+    let
+      prettyContents = case parseContract contents of
+        Right pcon -> show $ pretty pcon
+        Left _ -> contents
+    editorSetValue prettyContents
+    liftEffect $ LocalStorage.setItem marloweBufferLocalStorageKey prettyContents
+    --FIXME
+    -- updateContractInState prettyContents
+    -- resetContract
+    -- setOraclePrice settings
+    -- pure unit is only here so the comment stays in place
+    pure unit
+
+handleAction settings (SetEditorText contents) = do
+  editorSetValue contents
+  -- FIXME
+  -- updateContractInState contents
```

Comment:
actually I don't get what the FIXME here is?

Reply:
It was a placeholder for me to check later before removing the Draft, because we copy pasted some actions and I didn't investigate `updateContractInState` at that time. Now I can say it makes no sense in the MarloweEditor context so I removed it